File: app/src/main/java/com/housieshopping/app/domain/model/Banner.kt
package com.housieshopping.app.domain.model

data class Banner(
    val id: String,
    val title: String,
    val subtitle: String? = null,
    val imageUrl: String,
    val targetType: String, // e.g. "PRODUCT", "CATEGORY", "COUPON"
    val targetId: String
)

File: backend/src/models/Banner.js
import mongoose from 'mongoose';

const bannerSchema = new mongoose.Schema({
  title: { type: String, required: true },
  subtitle: { type: String, default: '' },
  imageUrl: { type: String, required: true },
  targetType: { type: String, enum: ['CATEGORY', 'PRODUCT', 'EXTERNAL'], default: 'CATEGORY' },
  targetId: { type: String, default: '' },
  categoryTarget: { type: String, default: '' },
  displayOrder: { type: Number, default: 0 },
  isActive: { type: Boolean, default: true, index: true }
}, {
  timestamps: true
});

export const Banner = mongoose.model('Banner', bannerSchema);
